java/compaction/compaction-job-execution/src/test/java/sleeper/compaction/job/execution/CompactionTaskTest.java

```diff
@@ -255,19 +272,137 @@ void shouldResetConsecutiveFailureCountIfJobProcessedSuccessfully() throws Excep
         }
     }
 
+    @Nested
+    @DisplayName("Update task status store")
+    class UpdateTaskStatusStore {
+        @Test
+        void shouldSaveTaskWhenOneJobSucceeds() throws Exception {
+            Queue<Instant> times = new LinkedList<>(List.of(
+                    Instant.parse("2024-02-22T13:50:00Z"), // Start
+                    Instant.parse("2024-02-22T13:50:01Z"), // Job completed
+                    Instant.parse("2024-02-22T13:50:02Z"), // Second check
+                    Instant.parse("2024-02-22T13:50:05Z"))); // Finish
+            createJobOnQueue("job1");
+
+            runTask("test-task-1", processJobs(
+                    jobSucceeds(summary(
+                            Instant.parse("2024-02-22T13:50:01Z"),
+                            Instant.parse("2024-02-22T13:50:02Z"), 10L, 10L))),
```

Comment:
Might it be worth reusing this summary object in the assertion? This applies to all the new tests.